apis/v1alpha1/gateway_types.go

```diff
@@ -425,21 +425,21 @@ const (
 // If NamespaceSelector and RouteSelector are defined, only routes matching both
 // selectors are associated with the Gateway.
 type RouteBindingSelector struct {
-	// RouteNamespaces indicates in which namespaces Routes should be selected
+	// Namespaces indicates in which namespaces Routes should be selected
 	// for this Gateway. This is restricted to the namespace of this Gateway by
 	// default.
 	//
 	// Support: Core
-	RouteNamespaces RouteNamespaces `json:"routeNamespaces"`
-	// RouteSelector specifies a set of route labels used for selecting
+	Namespaces RouteNamespaces `json:"namespaces"`
+	// Selector specifies a set of route labels used for selecting
 	// routes to associate with the Gateway. If RouteSelector is defined,
 	// only routes matching the RouteSelector are associated with the Gateway.
 	// An empty RouteSelector matches all routes.
 	//
 	// Support: Core
 	//
 	// +optional
-	RouteSelector metav1.LabelSelector `json:"routeSelector,omitempty"`
+	Selector metav1.LabelSelector `json:"selector,omitempty"`
```

Comment:
One potential negative here is that indenting `selector` one level could have a significant impact (would affect namespaces and not routes). I think this update is still a net improvement though, just wanted to mention it in case there were any other ideas.

Reply:
Yes, I realized that too. we could switch the ordering to help a little with that so instead of:

```yaml
      namespaces:
        from: "All"
      selector:
        matchLabels:
          "app": "foo"
```

do:

```yaml
      selector:
        matchLabels:
          "app": "foo"
      namespaces:
        from: "All"
```

That doesn't solve the problem but protects us a tiny bit. Most of this YAMLs will proliferate as is.

Reply:
👍 that's a great idea

Reply:
Updated all the yamls with this change as well.

Reply:
Thanks!